star-dash/star-dash/Persistence/EntityPersistable.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  EntityPersistable.swift
+//  star-dash
+//
+//  Created by Lau Rui han on 21/3/24.
+//
+
+import Foundation
+
+struct EntityPersistable: Encodable, Decodable {
+    var levelId: Int64
+    var entityType: EntityType
+    var position: CGPoint
+
+    init(levelId: Int64, entityType: EntityType, position: CGPoint) {
+        self.levelId = levelId
+        self.entityType = entityType
+        self.position = position
+    }
+
+    func toEntity() -> Entity {
+        switch entityType {
+        case .Monster:
+            return Monster(position: self.position)
+        case .Collectible:
+            return Collectible(position: self.position)
+        case .Obstacle:
+            return Obstacle(position: self.position)
+        case .Tool:
+            return Tool(position: self.position)
+        }
+    }
```

Comment:
This switch case can lead to cyclomatic complexity issues as we introduce more game entities, we could use a map of type [EntityType: () -> Entity] instead

Reply:
can you check if its done correctly

Reply:
I think the map could be static so we don't rebuild the map at every method call? That probably means the type should now be like [EntityType: (CGPoint) -> Entity]

Reply:
changed